pkg/transforms/mqttsecret.go

```diff
@@ -219,7 +260,7 @@ func (sender *MQTTSecretSender) MQTTSend(ctx interfaces.AppFunctionContext, data
 		},
 		tag)
 
-	if !sender.client.IsConnected() {
+	if !sender.client.IsConnected() && !sender.preConnected {
 		err := sender.connectToBroker(ctx, exportData)
```

Comment:
As the design of app-service will process each individual message in a separate goroutine, the scenario as described in https://github.com/edgexfoundry/app-functions-sdk-go/issues/1516 will have lots of goroutines attempting to make connection to MQTT broker here (mqtt broker is unreachable and sender.preConnected = false) and queueing for the mutex lock, which will cause abnormal memory soaring and leads to OOM.

Reply:
The added `!sender.preConnected` prevents the condition to be entered and avoids the queueing on the mutex lock. The  new ConnectToBroker must be used to avoid the queueing on the mutex lock.